drivers/amqp/broker.go

```diff
@@ -1,23 +1,131 @@
 package amqp
 
 import (
-	"fmt"
+	"log"
+
+	"github.com/google/uuid"
+	"github.com/streadway/amqp"
 
 	"github.com/go-celeriac/celeriac"
 )
 
 // AMQP is an instance of a celeriac Broker which uses amqp
-type AMQP struct{}
+type AMQP struct{
+	Connection amqp.Connection
```

Comment:
Any particular reason for not using a pointer here?

Reply:
No good reason